### ISSUE TITLE
Not authenticated - preview not available

### ISSUE THREAD

**Issue by kamerat** [sanity-io/next-sanity#38]
I'm trying to add the preview to our Sanity setup but when navigating to the preview url with cookies set, both sanity studio tab and the newly opened preview tab freezes. With further investigation it seems it freezes due to having an infinite loop trying to fetch /me:

![image](https://user-images.githubusercontent.com/11461961/149179837-97fe8fcb-1fa3-42b4-969d-8711ca40bf5f.png)

It seems to break on this line:
https://github.com/sanity-io/next-sanity/blob/d694ae3356bf29006aa2d0d0ef3a8bd96bd751ec/src/currentUser.ts#L16
![image](https://user-images.githubusercontent.com/11461961/149180286-24d4bd41-5625-41da-b9e4-2736d76b8988.png)

I've tested to navigate in the same browser to the url the function tries to fetch: https://github.com/sanity-io/next-sanity/blob/d694ae3356bf29006aa2d0d0ef3a8bd96bd751ec/src/currentUser.ts#L10
And this returns me successfully:
![image](https://user-images.githubusercontent.com/11461961/149181298-a3badcca-abc1-469c-8ae6-64d949dd9b02.png)

**What is causing this, and might it be related to #35? ( FYI: I'm using Google Chrome on Windows )**

**Comment by kamerat** [comment]
Been tooling with this issue many hours after following the guide https://www.sanity.io/guides/nextjs-live-preview.
I cloned the demo https://github.com/SimeonGriggs/sanity-nextjs-preview and got it to work with my local sanity setup. It turned out that I had done something in the FE part that Sanity did not like.

What fixed it for me was to refactor my FE to follow the cloned demo and then build it up again. My issue was most likely due to having multiple different sanity clients(multilang) and sharing the same config somehow.

**Comment by kamerat** [comment]
In case someone else sees this: I just wanted to add that I also used experimental spaces. This is likely the root of the problem here. The previewing works fine when running createPreviewSubscriptionHook syncronusly, but by wrapping it in a function with a dataset parameter, it makes things go mad. I never found ut if it was due to it being async or some mixup with the createClient config vs createPreviewSubscriptionHook config.. However, I ended up scrapping the experimental spaces and just switching between datasets manually.
